sale_delivery_date/models/sale_order_line.py

```diff
@@ -184,57 +349,72 @@ def _prepare_procurement_values_cutoff_time(
         the new planned date is delayed one day later. The argument
         keep_same_day forces keeping the same day.
         """
-        cutoff = self.order_id.get_cutoff_time()
-        partner = self.order_id.partner_shipping_id
         if not cutoff:
-            if not self.order_id.warehouse_id.apply_cutoff:
-                _logger.debug(
-                    "No cutoff applied on order %s as partner %s is set to use "
-                    "%s and warehouse %s doesn't apply cutoff."
-                    % (
-                        self.order_id,
-                        partner,
-                        partner.order_delivery_cutoff_preference,
-                        self.order_id.warehouse_id,
-                    )
-                )
-            else:
-                _logger.warning(
-                    "No cutoff applied on order %s. %s time not applied"
-                    "on line %s."
-                    % (self.order_id, partner.order_delivery_cutoff_preference, self)
-                )
-            return
-        new_date_planned = self._get_utc_cutoff_datetime(
-            cutoff, date_planned, keep_same_day
-        )
-        _logger.debug(
-            "%s applied on order %s. Date planned for line %s"
-            " rescheduled from %s to %s"
-            % (
-                partner.order_delivery_cutoff_preference,
-                self.order_id,
-                self,
-                date_planned,
-                new_date_planned,
+            return date_order
+        return self._get_utc_cutoff_datetime(cutoff, date_order, keep_same_day)
+
+    @api.model
+    def _postpone_to_working_day(self, date_start, calendar=False):
+        """Returns the nearest calendar's working day"""
+        if calendar:
+            # If inside an attendance, returns the nearest future working day
+            # at attendance start
+            return calendar.plan_hours(0, date_start, compute_leaves=True)
+        return date_start
+
+    @api.model
+    def _apply_customer_window(self, delivery_date, partner):
+        """Postpone a delivery date according to customer's delivery preferences"""
+        if partner.delivery_time_preference == "anytime":
+            return delivery_date
+        return partner.next_delivery_window_start_datetime(from_date=delivery_date)
+
+    @api.model
+    def _get_latest_work_end_from_date_range(
+        self, earliest_work_end, latest_expedition_date, calendar=False
+    ):
+        """Returns the nearest date in a calendar's attendance within a date range"""
+        if calendar:
```

Comment:
A bit easier to read:
```suggestion
        if not calendar:
           return latest_expedition_date
        ...
```